### ISSUE TITLE
Region goes behind taskbar

### ISSUE THREAD

**Issue by dhilgarth** [digitalcreations/MaxTo#119]
On my Windows 8.1 machine, the regions go behind the taskbar and thus, when maximizing, part of the window is behind the taskbar as well.
I am seeing this on my primary as well as my secondary monitor.

**Comment by vegardlarsen** [comment]
Thank you for reporting this. Is your taskbar set to auto-hide? If it is, MaxTo does not get told by Windows that the taskbar area is reserved (because it isn't).

Otherwise, it could be helpful if you run a tiny script I have to tell you the exact dimensions of your screen as reported by Windows. Do you happen to have LINQpad installed?

**Comment by dhilgarth** [comment]
No, my taskbar is not set to auto hide. LINQPad is installed, yes.

**Comment by vegardlarsen** [comment]
Could you quickly run this [extremely tiny script](https://dl.dropboxusercontent.com/u/12591686/MaxTo-diagnostics.linq) and get me a screenshot of the output?

**Comment by dhilgarth** [comment]
Here is the result: [Screens.xlsx](https://github.com/digitalcreations/MaxTo/files/52057/Screens.xlsx)

**Comment by vegardlarsen** [comment]
According to that, you can see that the `WorkingArea` size (the size subtracting the taskbar) is the same as the `Bounds` size, which means that Windows is incorrectly reporting that size on your machine.

There really is nothing MaxTo can do about this. I'd recommend googling the issue, and seeing if you can get rid of the problem. You can easily verify that this happens when MaxTo is not running as well, simply uncheck "Start with Windows" in MaxTo's options and sign out of you computer.

**Comment by dhilgarth** [comment]
It's not happening without MaxTo: http://screencast.com/t/g2VRFKqfIi

**Comment by vegardlarsen** [comment]
That is extremely weird. Do the `WorkingArea` sizes reported by the LINQpad script change when MaxTo is not running?

**Comment by dhilgarth** [comment]
No, there is no difference with or without MaxTo

**Comment by vegardlarsen** [comment]
It could be a video driver issue. Could you try updating to your latest drivers?

Note that the LINQpad script does not reference MaxTo at all, as I am sure you have discovered. MaxTo only reads those sizes from Windows and uses them to set up the regions. If they are wrong, there isn't much else we can do.

**Comment by vegardlarsen** [comment]
You can try `System.Windows.SystemParameters.WorkArea.Height.Dump()` and comparing it to the working area size reported for your primary monitor by `Screen.AllScreens`?

**Comment by dhilgarth** [comment]
That returns the correct value: 1018

**Comment by vegardlarsen** [comment]
Could you try unlocking your taskbar and putting it on one side? Does that trigger the the LINQpad script to get updated numbers?

**Comment by dhilgarth** [comment]
Nope, no difference.

**Comment by vegardlarsen** [comment]
I don't really understand what is happening here. I noticed your monitors were labelled `\.\\DISPLAY4` and `5`, skipping a few numbers. No. 4 would be your primary monitor?

**Comment by vegardlarsen** [comment]
Do any of your monitors happen to be high-DPI displays?

**Comment by dhilgarth** [comment]
I don't know, why a few numbers are skipped. No High DPI displays, just two Samsung SyncMaster SA350 21.5" screens.

**Comment by vegardlarsen** [comment]
It seems your Windows installation has gone insane :)

I have to leave now, but a few suggestions:

- After moving your taskbar, ensure you close LINQpad between tries, because I have a suspicion the output gets cached.
- Try rebooting?

**Comment by AssaultCommand** [comment]
A bit late now, but I've been having the exact same issue with the only exception being that if I restart MaxTo it works perfectly fine again.

Running Windows 10 on my desktop with a 3440x1440 monitor. It also happens on my laptop whether I plug it into the monitor or not.

However, I can't exactly test MaxTo right now because the trials ran out and I just came across this thread today.

**Comment by brettw1234** [comment]
@AssaultCommand I am also running Windows 10 on a 3440x1440 monitor. I have been having the exact same issues until I found this thread and decided to close the application and re-open it - which magically fixed it. Since I installed it several months ago, the application has always started when Windows booted up and I never manually closed/re-opened it. Have you tried that?

**Comment by AssaultCommand** [comment]
Have since bought a license because almost all alternatives suck. 

I pretty much never reboot my desktop but I don't think it's that causing it. It usually happens after I wake it from sleep. Might have something to do with the bug in Windows where it shrinks all your windows to the smallest monitor resolution supported and puts them all in the top left corner.

**Comment by brettw1234** [comment]
Interesting - I almost never reboot my desktop either and just let it go to sleep. After the restart and closing/re-opening the application yesterday I haven't been able to reproduce the issue. I'll keep an eye out though and will post back if I ever find a way to reproduce it.

**Comment by AssaultCommand** [comment]
I'd like to propose a reset button/restart button on the context menu on the icon in the tray. Would solve most of my issues in the shortest amount of time.

**Comment by AssaultCommand** [comment]
I'd like to propose a reset button/restart button on the context menu on the icon in the tray. Would solve most of my issues in the shortest amount of time. @vegardlarsen

**Comment by vegardlarsen** [comment]
First, I apologize for the slow reply. I have been on vacation for the first time in a long time, so I haven't responded to anything in quite a while.

The problem here is that a reset button doesn't work in these cases. MaxTo is asking Windows to please notify it whenever the available working space changes, which it usually does just fine in most cases.

It seems that in the cases where this happens, MaxTo is getting the wrong sizes from Windows, and just "resetting" doesn't work. One would have to actually close down MaxTo to get the new, corrected sizes. I don't know why this happens, but I am looking at solutions.

**Comment by AssaultCommand** [comment]
Sorry, with the restart button I meant that it'd actually restart maxto instead of some sort of reset as a manual restart of the program had always fixed it for me so far.

**Comment by AlexanderLiuckanov** [comment]
Hi,

I have submitted feedback to Microsoft for this problem. if you are a Windows 10 Insider, go to the feedback hub and vote for this problem. 
Category: Developer Platform
Subcategory: Navigation, views, and windows
Title: incorrect result from System.Windows.Forms.Screen.PrimaryScreen.WorkingArea.Height

**Comment by datune** [comment]
Restarting MaxTo does not help at all, at least in my case. I have the exact same issue as OP, Windows 10 Pro...

**Comment by EntuTeque** [comment]
It has been nearly a year since anyone commented on this. Wondering if there has been any development. I seem to have this problem occasionally. Its no small matter as I like my taksbar at the top, so windows that maximize under the taskbar suddenly have their min-max-close buttons inaccessible.

**Comment by AssaultCommand** [comment]
Nope, still _patiently_ waiting on a fix here. Doubt it's coming.

**Comment by vegardlarsen** [comment]
It will most likely be fixed shortly. The problem here is that this appears to be a bug in .NET Framework (specifically caching of `Screen.WorkingArea` without proper invalidation).

In the next release of MaxTo, we will no longer relying on this class at all.

**Comment by NKame** [comment]
Same problem here: windows regularly go under the taskbar (not configured for auto-hide), no high DPI display, no zoom factor. (And there is a weird case of windows, typically web browser playing a video full screen, switching monitor when exiting full screen, but that's for another ticket).

**Comment by AssaultCommand** [comment]
Yep, still having the issue to this day too.

**Comment by vegardlarsen** [comment]
"Shortly" was a bit longer than I expected. We are no longer relying on the `System.Windows.Forms.Screen` implementation in the 2.0.0-preview releases that you can read about in issue #220. Can someone experiencing this problem please try out the preview release and let me know if the issue disappears?

I am closing this issue in the expectation that this has been fixed in the 2.0.0-releases. Please let me know if I was mistaken.
